### PR TITLE
feat: update to net8 and C#12

### PR DESCRIPTION
**Pull request by carlh98**
GitHub Issue: #

## Proposed Changes

 - [ ] Bug fix
 - [x] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

Update to net8 and C#12

## Impact on version

- [ ] **Major**
  - The template structure was changed.
- [x] **Minor**
  - New functionalities were added.
- [ ] **Patch**
  - A bug in behavior was fixed.
  - Documentation was changed.

### Always applicable
No matter your changes, these checks always apply.
- [x] Your conventional commits are aligned with the **Impact on version** section.
- [x] Updated [CHANGELOG.md](../CHANGELOG.md).
  - Use the latest Major.Minor.X header if you do a **Patch** change.
  - Create a new Major.Minor.X header if you do a **Minor** or **Major** change.
  - If you create a new header, it aligns with the **Impact on version** section and matches what is generated in the build pipeline.
- [x] Documentation files were updated according with the changes.
  - Update `README.md` and `TemplateConfig.md` if you made changes to **templating**.
  - Update `AzurePipelines.md` and `APP_README.md` if you made changes to **pipelines**.
  - Update `Diagnostics.md` if you made changes to **diagnostic tools**.
  - Update `Architecture.md` and its diagrams if you made **architecture decisions** or if you introduced new **recipes**.
  - ...and so forth: Make sure you update the documentation files associated to the recipes you changed. Review the topics by looking at the content of the `doc/` folder.
- [x] Images about the template are referenced from the [wiki](https://github.com/nventive/UnoApplicationTemplate/wiki/Images) and added as images in this git.
- [x] TODO comments are hints for next steps for users of the template and not planned work.

### Contextual
Based on your changes these checks may not apply.
- [ ] Automated tests for the changes have been added/updated.
- [ ] Tested on all relevant platforms

### CODE REVIEW

File: src/app/ApplicationTemplate.Windows/Properties/Properties/win10-x86.pubxml [hunk 1]
```diff
@@ -6,7 +6,7 @@ https://go.microsoft.com/fwlink/?LinkID=208121.
   <PropertyGroup>
     <PublishProtocol>FileSystem</PublishProtocol>
     <Platform>x86</Platform>
-    <RuntimeIdentifier>win10-x86</RuntimeIdentifier>
```
Comment (Soap-141) [inline src/app/ApplicationTemplate.Windows/Properties/Properties/win10-x86.pubxml:9]: Should we rename the specific files for this too?

**Review by LProulx86 — APPROVED** [review]

File: src/app/ApplicationTemplate.Windows/Properties/Properties/win10-x86.pubxml (hunk 1 above)
Comment (carlh98) [reply src/app/ApplicationTemplate.Windows/Properties/Properties/win10-x86.pubxml:9]: Good question.

@LProulx86 can you check this

File: src/app/ApplicationTemplate.Windows/ApplicationTemplate.Windows.csproj [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 		<TargetPlatformMinVersion>10.0.19041.0</TargetPlatformMinVersion>
 		<SupportedOSPlatformVersion>10.0.20348.0</SupportedOSPlatformVersion>
 		<RootNamespace>ApplicationTemplate</RootNamespace>
 		<GenerateDocumentationFile>true</GenerateDocumentationFile>
 		<ApplicationManifest>app.manifest</ApplicationManifest>
 		<Platforms>x86;x64;arm64</Platforms>
-		<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
-		<PublishProfile>win10-$(Platform).pubxml</PublishProfile>
+		<RuntimeIdentifiers>win-x86;win-x64;win-arm64</RuntimeIdentifiers>
+		<SelfContained>true</SelfContained>
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Windows/ApplicationTemplate.Windows.csproj:13]: Is this supposed to be here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,8 +5,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
 Prefix your items with `(Template)` if the change is about the template and not the resulting application.
 
+## 3.1.X
+- Updated to .NET 8.
+
 ## 3.0.X
-- Updated to Uno 5.
+- Updated to Uno 5 and .NET 7.
```
[the remark this answers is not in the document]
Comment (carlh98) [reply CHANGELOG.md:12]: true, my mistake

File: src/app/ApplicationTemplate.Windows/ApplicationTemplate.Windows.csproj (hunk 2 above)
Comment (carlh98) [reply src/app/ApplicationTemplate.Windows/ApplicationTemplate.Windows.csproj:13]: yes, they made a change to net8, self contained was used by default previously
https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/8.0/runtimespecific-app-default

File: src/app/ApplicationTemplate.Windows/Properties/Properties/win10-x86.pubxml (hunk 1 above)
Comment (LProulx86) [reply src/app/ApplicationTemplate.Windows/Properties/Properties/win10-x86.pubxml:9]: Fix. remove win10 for win

File: src/app/ApplicationTemplate.Windows/ApplicationTemplate.Windows.csproj (hunk 2 above)
Comment (jeanplevesque) [reply src/app/ApplicationTemplate.Windows/ApplicationTemplate.Windows.csproj:13]: Ok, but should we keep SelfContained if it's no longer the default? Is it building faster without it?

Comment (jeanplevesque) [reply src/app/ApplicationTemplate.Windows/ApplicationTemplate.Windows.csproj:13]: Actually, let's keep it like this for now to limit the changes. But we should consider staying on the default value.

**Review by jeanplevesque — APPROVED** [review]

**Review by Soap-141 — APPROVED** [review]

**Pull request merged**